### PR TITLE
Adds a new App for Modular Computers, the NTOS Arcade.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
Hey you, punk Are you hip with modular computers?
...What's that, you think there's nothing to do on that old tablet of yours? Whoah brochacho, don't dump that tablet in the trash, and check out the new _gaming app_ that's sweeping the station!

## About The Pull Request
Finally starts adding some more functionality to modular computers that people might actually use with the NTOS Arcade App. It's as close to a full port of the Regular Arcade app as I could manage, I'm not sure about how I implemented the spam prevention, as without it you can buffer multiple attacks at once, but this way the game plays pretty much the same. In addition, the civilian modular console comes pre-installed with the app, to give crew a reason to check it out. complete with  the same damage values, sound effects, and wrapped in a nice modular computer UI Shell. There's another PR by StyleMistake right now that'll allow me to apply the new UI style used in other, nicer apps here as well.
NTOSArcade

In addition, winning the arcade game via a modular app prints out tickets as opposed to dispensing a prize directly. In order to claim a prize, deposit 2 tickets into another, free-standing arcade to earn a random prize, as well as applying the usual arcade mood buff.
image

## Why It's Good For The Game
Modular Computers are cool, and this was quite literally my first idea for how to get more people to use them as a whole. It might encourage people to use their tablets for something other then contracts, sneaking ID consoles, and SM monitoring.

## Changelog
:cl:
add: Nanotrasen's software development team has finally finished a port if their hit, battle arcade game, available now on tablets, laptops, and modular computers.
add: Printer-enabled modular computers are now capable of printing arcade tickets, which can be redeemed for regular prizes at standard arcade computers.
/:cl:

### CODE REVIEW

File: code/modules/modular_computers/file_system/programs/arcade.dm [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	var/boss_mp = 15
+	var/player_hp = 30
+	var/player_mp = 10
+	var/ticket_count = 0
+	var/heads_up = "Nanotrasen says, winners make us money."//Shows the active display text for the app
+
+/datum/computer_file/program/arcade/proc/game_check(mob/user)
+	sleep(5)
+	if(boss_hp <= 0)
+		heads_up = "You have crushed the forces of evil! Rejoyce!"
```
Comment (GravitronBlue) [inline code/modules/modular_computers/file_system/programs/arcade.dm]: Rejoyce?

File: code/modules/modular_computers/file_system/programs/arcade.dm [hunk 2]
```diff
@@ -0,0 +1,166 @@
+/datum/computer_file/program/arcade
+	filename = "arcade"
+	filedesc = "Nanotransen Micro Arcade"
+	ui_header = "smmon_0.gif"
```
Comment (stylemistake) [inline code/modules/modular_computers/file_system/programs/arcade.dm]: That will draw an SM monitor icon in the header, I think you don't want that.

File: code/modules/modular_computers/file_system/programs/arcade.dm (hunk 1 above)
Comment (ArcaneMusic) [reply code/modules/modular_computers/file_system/programs/arcade.dm]: It's not one of my PRs without at LEAST one spelling error.

**Comment by kriskog** [comment]
Okay, how do I emag it?

**Review by stylemistake — DISMISSED** [review]

**Comment by stylemistake** [comment]
Theme aside, this PR looks good.

**Comment by Tharcoonvagh** [comment]
Finally, I was about to make a thread about tablet games

File: code/modules/modular_computers/file_system/programs/arcade.dm [hunk 3]
```diff
@@ -0,0 +1,165 @@
+/datum/computer_file/program/arcade
+	filename = "arcade"
+	filedesc = "Nanotransen Micro Arcade"
```
Comment (stylemistake) [inline code/modules/modular_computers/file_system/programs/arcade.dm]: Nano**transen** 👀

Comment (Quartich) [reply code/modules/modular_computers/file_system/programs/arcade.dm]: Nanotrasen is most popular spelling, but I have heard it a few ways

Comment (ArcaneMusic) [reply code/modules/modular_computers/file_system/programs/arcade.dm]: It's not an ArcaneMusic PR without at LEAST one misspelling.

Comment (Tharcoonvagh) [reply code/modules/modular_computers/file_system/programs/arcade.dm]: Feature

**Comment by Tlaltecuhtli** [comment]
i like oregon trial more

**Comment by Farquaar** [comment]
Can I buy in-game DLC with my in-game currency?

**Comment by ArcaneMusic** [comment]
> Will this app feature microtransactions?

Paypal me 5 dollars and find out!

**Comment by ATH1909** [comment]
Do you earn tickets at the same "rate" with the mobile game as you do with the stationary arcade machines?

**Comment by ArcaneMusic** [comment]
> Do you earn tickets at the same "rate" with the mobile games as you do with the stationary arcade machines?

If you win one game on the app then you earn one ticket, but you need 2 tickets to earn a prize, unlike the 1 win-> 1 prize on regular arcades,  just because of how much more available the app is.

**Comment by stylemistake** [comment]
I think this PR still needs some work, mainly - the boss image doesn't load for some reason, and you can stack multiple actions by pressing them quickly (which i thought you solved long time ago with locks).

**Comment by ArcaneMusic** [comment]
Well the good news is I got the icons working! It was as simple as me ignoring to have .gif in the src of the  label.
image
There's now 5 different icons available for use. I had also tried a few times to get the randomized boss names from the regular arcade working in the same way, but I was running into some issues with pathing to the .json arcade file so this will do for now anyway. Also ALSO I didn't realize that when I was implementing the spam prevention, I prevented spamming by the player, but NOT the boss's actions. This should be fixed now after I did some testing.

**Comment by ArcaneMusic** [comment]
Alright, I touched up a few of sprites that I literally recycled to give them some more flavor, and took some advice from the discord to shift around the UI and make it look a lot more compact, but certainly a lot more game-y. Each player bar also has coloring to show when you're overhealed/charged, so all together I'd say is in a state I'm happy to leave it in.

**Comment by gwhit94** [comment]
I don't like that I need two tickets to win a prize. Having to win twice is tedious, especially since it isn't Orion Trail.

**Pull request merged**